src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -690,19 +696,43 @@ private void setNotificationPriority(Bundle extras, NotificationCompat.Builder m
         }
     }
 
+    private Bitmap getCircleBitmap(Bitmap bitmap) {
+        final Bitmap output = Bitmap.createBitmap(bitmap.getWidth(), bitmap.getHeight(), Bitmap.Config.ARGB_8888);
+        final Canvas canvas = new Canvas(output);
+        final int color = Color.RED;
+        final Paint paint = new Paint();
+        final Rect rect = new Rect(0, 0, bitmap.getWidth(), bitmap.getHeight());
+        final RectF rectF = new RectF(rect);
+
+        paint.setAntiAlias(true);
+        canvas.drawARGB(0, 0, 0, 0);
+        paint.setColor(color);
+        canvas.drawOval(rectF, paint);
+
+        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
+        canvas.drawBitmap(bitmap, rect, rect, paint);
+
+        bitmap.recycle();
+
+        return output;
+    }
+
     private void setNotificationLargeIcon(Bundle extras, String packageName, Resources resources, NotificationCompat.Builder mBuilder) {
         String gcmLargeIcon = extras.getString(IMAGE); // from gcm
         if (gcmLargeIcon != null && !"".equals(gcmLargeIcon)) {
             if (gcmLargeIcon.startsWith("http://") || gcmLargeIcon.startsWith("https://")) {
-                mBuilder.setLargeIcon(getBitmapFromURL(gcmLargeIcon));
+                Bitmap bitmap = getBitmapFromURL(gcmLargeIcon);
+                Bitmap bm = getCircleBitmap(bitmap);
+                mBuilder.setLargeIcon(bm);
                 Log.d(LOG_TAG, "using remote large-icon from gcm");
             } else {
                 AssetManager assetManager = getAssets();
                 InputStream istr;
                 try {
                     istr = assetManager.open(gcmLargeIcon);
                     Bitmap bitmap = BitmapFactory.decodeStream(istr);
-                    mBuilder.setLargeIcon(bitmap);
+                    Bitmap bm = getCircleBitmap(bitmap);
+                    mBuilder.setLargeIcon(bm);
                     Log.d(LOG_TAG, "using assets large-icon from gcm");
```

Comment:
Cool. I'm fine with making it an option. 

As far back as I've had Android devices and my memory serves, not sure which is longer, all large notifications have been round. The revision you have above is very similar to what I had at commit `9c351c2`. I was just having trouble getting the condition to catch that I had passed `roundIcon: true` to it. Would you want to see the condition set in `init` as @fr` suggested? Or do you like it as part of the ` data` payload?

Regarding copyright, I hadn't asked permission. Some experience with a former cofounder of mine led me into a discussion with a patent attorney. Basically only creative works are defensible as copyright. I wouldn't say something as standard as creating a circular image from a square one with an API falls within that realm. But I'm happy to post a question on the blog post and put a screenshot of the author's reply here if you like.